src/binaryutils.cpp

```diff
@@ -258,12 +186,64 @@ Status validateTensor(const std::shared_ptr<TensorInfo>& tensorInfo,
     return StatusCode::OK;
 }
 
+Dimension getTensorInfoHeightDim(const std::shared_ptr<TensorInfo>& tensorInfo) {
+    size_t numberOfShapeDimensions = tensorInfo->getShape().size();
+    if (numberOfShapeDimensions < 4 || numberOfShapeDimensions > 5) {
+        throw std::logic_error("wrong number of shape dimensions");
+    }
+    size_t position = numberOfShapeDimensions == 4 ? /*NHWC*/ 1 : /*N?HWC*/ 2;
+    return tensorInfo->getShape()[position];
+}
+
+Dimension getTensorInfoWidthDim(const std::shared_ptr<TensorInfo>& tensorInfo) {
+    size_t numberOfShapeDimensions = tensorInfo->getShape().size();
+    if (numberOfShapeDimensions < 4 || numberOfShapeDimensions > 5) {
+        throw std::logic_error("wrong number of shape dimensions");
+    }
+    size_t position = numberOfShapeDimensions == 4 ? /*NHWC*/ 2 : /*N?HWC*/ 3;
+    return tensorInfo->getShape()[position];
+}
+
+void updateAlignmentResolution(Dimension& height, Dimension& width, const cv::Mat& image) {
+    if (height.isAny()) {
+        height = image.rows;
+    } else if (height.isDynamic()) {
+        if (height.match(image.rows)) {
+            height = image.rows;
+        } else {
+            if (image.rows > height.getMaxValue()) {
+                height = height.getMaxValue();
+            } else {
+                height = height.getMinValue();
+            }
+        }
+    }
+    if (width.isAny()) {
+        width = image.cols;
+    } else if (width.isDynamic()) {
+        if (width.match(image.cols)) {
+            width = image.cols;
+        } else {
+            if (image.cols > width.getMaxValue()) {
+                width = width.getMaxValue();
+            } else {
+                width = width.getMinValue();
+            }
+        }
+    }
+}
+
 Status convertTensorToMatsMatchingTensorInfo(const tensorflow::TensorProto& src, std::vector<cv::Mat>& images, const std::shared_ptr<TensorInfo>& tensorInfo) {
+    Dimension tensorInfoHeightDim = getTensorInfoHeightDim(tensorInfo);
+    Dimension tensorInfoWidthDim = getTensorInfoWidthDim(tensorInfo);
+
     for (int i = 0; i < src.string_val_size(); i++) {
         cv::Mat image = convertStringValToMat(src.string_val(i));
         if (image.data == nullptr)
             return StatusCode::IMAGE_PARSING_FAILED;
 
+        updateAlignmentResolution(tensorInfoHeightDim, tensorInfoWidthDim, image);
```

Comment:
getAlignedResolution?

Reply:
This updates the resolution, therefore `getAlignedResolution` would be misleading

Reply:
`updateTargetResolution`

Reply:
Done